pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -395,6 +395,10 @@ func (r *repository) getImageOfImageStream(dgst digest.Digest) (*imageapi.Image,
 		return nil, nil, wrapKStatusErrorOnGetImage(r.name, dgst, err)
 	}
 
+	// We are going to pretend that we've got the image from the same source
+	// as the client.
+	image.DockerImageReference = tagEvent.DockerImageReference
```

Comment:
I found only one place where it was possible. Now it's fixed and covered by the test.